includes/apple-exporter/components/class-body.php

```diff
@@ -440,6 +439,51 @@ public function set_default_style() {
 		);
 	}
 
+	/**
+	 * Determine whether to apply a dropcap style for the component.
+	 * 
+	 * @param \Apple_Exporter\Theme $theme Object that stores theme level dropcap configuration.
+	 * @param string                $html The HTML to check for dropcap conditions. Should be the first paragraph of the post content.
+	 *
+	 * @return boolean
+	 */
+	private function dropcap_determination( $theme, $html ) {
+		// Toggle dropcap determination flag so that this logic applies only to the post's first paragraph.
+		$theme->dropcap_applied = true;
+
+		// Check that the theme is configured to apply dropcap styling.
+		if ( 'yes' !== $theme->get_value( 'initial_dropcap' ) ) {
+			return false;
+		}
+
+		// Check the first character for punctuation.
+		$content    = wp_strip_all_tags( $html );
+		$first_char = mb_substr( $content, 0, 1 );
+		// Regex-planation: \p{P} searchs for punctuation, /u modifier makes it unicode inclusive.
+		if ( preg_match( '/\p{P}$/u', $first_char ) ) {
+			return false;
+		}
+
+		// Check that the content meets the minimum character number.
+		$num_chars = mb_strlen( $content );
+		if ( 'yes' !== $theme->get_value( 'dropcap_minimum_opt_out' )
+			&& $num_chars < (int) $theme->get_value( 'dropcap_minimum' )
+		) {
+			return false;
+		}
+
+		/** 
+		 * Allows for filtering of the dropcap content before return.
+		 * 
+		 * @since 2.4.0
+		 * 
+		 * @param string                $content The content to filter.
+		 * @param \Apple_Exporter\Theme $theme The theme whose dropcap options are used.
+		 * @param string                $post_id The id of the post whose content we're parsing.
+		 */
+		return apply_filters( 'apple_news_dropcap', $content, $theme, $this->workspace->content_id ); 
```

Comment:
This is close, but needs to pass the determination as the first parameter, and the `return` values above need to be set into a variable instead. Assuming the variable is called `$use_dropcap`:
```suggestion
		/** 
		 * Filters the automatic determination of whether to use a dropcap on a paragraph or not.
		 * 
		 * @since 2.4.0
		 * 
		 * @param bool                  $use_dropcap Whether to use a dropcap for this paragraph or not.
		 * @param string                $content     The content to filter.
		 * @param \Apple_Exporter\Theme $theme       The theme whose dropcap options are used.
		 * @param int                   $post_id     The id of the post whose content we're parsing.
		 */
		return apply_filters( 'apple_news_dropcap', $use_dropcap, $content, $theme, $this->workspace->content_id ); 
```